### ISSUE TITLE
`lean --stdin --run` ignores the first trailing argument

### ISSUE THREAD

**Issue by Kha** [leanprover/lean4#4601]
```
$ echo 'def main (as : List String) : IO Unit := IO.println (repr as)' | lean --stdin --run '«path with spaces»' '«path with spaces»'
["«path with spaces»"]
$ lean <(echo 'def main (as : List String) : IO Unit := IO.println (repr as)') --run '«path with spaces»' '«path with spaces»'
["«path with spaces»", "«path with spaces»"]
```

**Comment by Julian** [comment]
I was about to open another issue for `lean --stdin` behavior, but given it's somewhat similar and that I'm not sure whether it's really fixable, I'll post it here unless someone tells me otherwise. `lean --stdin` also exits successfully and silently when given "nonexisting" arguments.

Specifically:

```
$ ⊙  printf 'def main : IO Unit := IO.println "Hello, world!"\n' | lean run --stdin
```

exits with exit code 0. Note that I typoed `lean run` for `lean --run`.

It seems not 100% trivial to suggest a fix, given that I presume the way things work now are meant to facilitate someone who's passing arguments to the Lean code they're running. ISTM that ideally `--stdin` wouldn't allow arguments *before* itself, only after, or only perhaps after `--`, but I'm not sure how much that'd complicate the C++ argument parsing if at all.
